wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -216,6 +216,19 @@ abstract class Wallet
     }
   }
 
+  private def searchFilterMatches(spks: Vector[ScriptPubKey])(
+      blockFilters: Vector[(DoubleSha256Digest, GolombFilter)]): Future[
+    Vector[DoubleSha256Digest]] = Future {
```

Comment:
nit: this kind of feels like it should be an Option instead of a Vector. If you really want to use flatten above you can still call `toVector` on the returned Options

Reply:
It would have to be an `Option[Vector[DoubleSha256Digest]]` then, i guess i generally prefer just doing a `Vector`. Note the `blockFilters.flatMap` call below.